contracts/examples/multisig/tests/multisig_rust_test_v2.rs

```diff
@@ -0,0 +1,319 @@
+use adder::ProxyTrait as AdderProxyTrait;
+use elrond_wasm::{
+    elrond_codec::multi_types::{MultiValueVec, OptionalValue},
+    storage::mappers::SingleValue,
+    types::{Address, CodeMetadata},
+};
+use elrond_wasm_debug::{
+    mandos::{
+        interpret_trait::{InterpretableFrom, InterpreterContext},
+        model::{
+            Account, AddressKey, AddressValue, ScCallStep, ScDeployStep, SetStateStep, TxExpect,
+        },
+    },
+    BlockchainMock, ContractInfo, DebugApi,
+};
+use multisig::{
+    multisig_perform::ProxyTrait as MultisigPerformProxyTrait,
+    multisig_propose::ProxyTrait as MultisigProposeProxyTrait, ProxyTrait as MultisigProxyTrait,
+};
+use num_bigint::BigUint;
+
+type MultisigContract = ContractInfo<multisig::Proxy<DebugApi>>;
+type AdderContract = ContractInfo<adder::Proxy<DebugApi>>;
+
+struct Users {
+    alice: Address,
+    bob: Address,
+    carol: Address,
+}
+
+fn world() -> BlockchainMock {
+    let mut blockchain = BlockchainMock::new();
+    blockchain.set_current_dir_from_workspace("contracts/examples/multisig");
+
+    blockchain.register_contract_builder("file:test-contracts/adder.wasm", adder::ContractBuilder);
+    blockchain.register_contract_builder("file:output/multisig.wasm", multisig::ContractBuilder);
+    blockchain
+}
+
+#[test]
+fn basic_setup_test() {
+    let _ = DebugApi::dummy();
+
+    let world = &mut world();
+    let ic = &world.interpreter_context();
+
+    let users = setup_users(world, ic);
+
+    let multisig = &mut multisig_deploy(&users, world, ic);
+
+    let board_members: MultiValueVec<Address> = world.mandos_sc_call_get_result(
+        multisig.get_all_board_members(),
+        ScCallStep::new()
+            .from(users.alice.as_array())
+            .expect(TxExpect::ok()),
+    );
+
+    let expected_board_members: Vec<_> = [users.alice, users.bob, users.carol].into();
+
+    assert_eq!(board_members.into_vec(), expected_board_members);
+}
+
+#[test]
+fn multisig_adder_test() {
+    let _ = DebugApi::dummy();
+
+    let world = &mut world();
+    let ic = &world.interpreter_context();
+
+    let users = setup_users(world, ic);
+    let source_adder = &mut adder_deploy(world, ic);
+    let multisig = &mut multisig_deploy(&users, world, ic);
+    let caller = &users.alice;
+    let signers = [&users.alice, &users.bob, &users.carol];
+    let adder = &mut multisig_deploy_adder(
+        multisig,
+        source_adder,
+        caller,
+        signers.as_slice(),
+        world,
+        ic,
+    );
+    let expected_adder_address = AddressKey::interpret_from("sc:adder-multisig", ic);
+    assert_eq!(
+        adder.mandos_address_expr.value,
+        expected_adder_address.value
+    );
+
+    let first_value: BigUint = 42u64.into();
```

Comment:
Couldn't we actually only work with u64 or even u32 up to the contract proxy? There will unlikely be any big numbers in the test.